### ISSUE TITLE
Valve should not open when battery runs out of power

### ISSUE THREAD

**Issue by magicdude4eva** [EUROTRONIC-Technology/Spirit-ZigBee#4]
As the battery runs low / out of power, the valve completely opens/resets. I struggle reliably testing this, but noticed it with two TRVs which were turned off, they completely opened. My suggestion would be:
a) Close the valve completely if battery level drops below 10% - then it is up to the user to override/replace
b) Keep the valve position and do not reset when battery level drops

**Comment by gitolicious** [comment]
This is a tricky one. With a closed valve it can be way harder to remove the thermostat. In case you don't have fresh batteries at hand you are stuck with a cold heater with no chance to replace the thermostat with a "dumb" one. With an open valve you might waste a lot of heating energy if you are not around to replace the batteries quickly. Both not ideal, but unsure which one would be better.

**Comment by magicbenny** [comment]
i think just in case of a leak and empty batteries a closed valve would be better than an open one or am i wrong? 😁

**Comment by magicdude4eva** [comment]
@gitolicious replacing the Eurotronic with the valve closed is no different than with the valve open. A open valve is certainly more risk (leakage, waste of energy).

**Comment by florianschroen** [comment]
Another aspect is the used battery type.
When using NiMH rechargeable batteries, the reference voltage is 1.2V instead of the normal 1.5V which results in quickly appearing battery low warnings.

When actively shutting down / disabling the TRV at fixed voltage, there is much remaining runtime and energy "lost".

I had a TRV which had a problem with lost connections. With rechargeable batteries the TRV powered off after only 3 weeks!

The newly ordered one is running with battery low warning since few days.

Maybe disabling this active power off feature completely would be an option? (which seems equal to keep the last valve position)

**Comment by dschulte** [comment]
> @gitolicious replacing the Eurotronic with the valve closed is no different than with the valve open. A open valve is certainly more risk (leakage, waste of energy).

A closed valve makes absolutely no difference in case of a leak as the radiator is always connected to the loop through the outlet. The valve should be open with drained batteries as a closed valve could result in frozen water which will result in much more damage than the wasted energy.

**Comment by magicbenny** [comment]
> > @gitolicious replacing the Eurotronic with the valve closed is no different than with the valve open. A open valve is certainly more risk (leakage, waste of energy).
> 
> 
> 
> The valve should be open with drained batteries as a closed valve could result in frozen water which will result in much more damage than the wasted energy.

a radiator valve is never fully closed du tue exactly the reason you mentioned...freezing. so it really doesn't matter what state the valve has when batteries are low or empty.

**Comment by dschulte** [comment]
> > > @gitolicious replacing the Eurotronic with the valve closed is no different than with the valve open. A open valve is certainly more risk (leakage, waste of energy).
> > 
> > 
> > The valve should be open with drained batteries as a closed valve could result in frozen water which will result in much more damage than the wasted energy.
> 
> a radiator valve is never fully closed du tue exactly the reason you mentioned...freezing. so it really doesn't matter what state the valve has when batteries are low or empty.

No, this behavior is part of the thermostat. A closed valve has no flow under normal circumstances. You can proof it by closing the valve and the radiator will be completely cold after some time. A common thermostat has a frost protection which will hold a minimum temperature of 6°C when turned completely to the off position.

**Comment by magicdude4eva** [comment]
> a radiator valve is never fully closed du tue exactly the reason you mentioned...freezing. so it really doesn't matter what state the valve has when batteries are low or empty.

This is not true all radiators can be completely shutoff mechanically via the valve. With a "mechanical" thermostat freezing can be prevented by putting the thermostat into the "star"-position. 

TBH: When fitting a thermostatic radiator valve, especially a digital/smart thermostat like the Eurotronic, I expect it to assist me controlling the temperature perfectly - be it protecting against freezing or regulating the temperature controlling it perfectly. It should have like any other system a fallback/emergency mode.

If a complete shutdown/closing the valve is considered to be a risk, I would recommend leaving the valve in "frost-mode" - slightly open.

**Comment by gitolicious** [comment]
As seen by the discussions, there is no simple "best way" for the manufacturers to implement.

 > I would recommend leaving the valve in "frost-mode" - slightly open

This will not be easily achievable. There is no correct "slightly open" in a sense of frost protection which is valid for all installations as it depends heavily on external factors. Setting a classical thermostat to "star"/snowflake does not "slightly open" the thermostat - it simply sets the target temperature to 6 °C and opens the valve if the temperature drops below that value.

**Comment by magicbenny** [comment]
to change the point of view a bit i am asking myself why this is so important? due to the fact that we get the battery state as a sensor in the smart home tool of our choice it is possible to built some automations. so what is so difficult to act when the battery state is aprox. round about 10% and simply recharge or change them? if doing so no need for making things more complicate then necessary.

**Comment by florianschroen** [comment]
That's a good point of view.

But this would depend on 
a) a working smarthome infrastructure 
b) a correct battery state

When inserting fully charged batteries, the battery level is at 15%.
10% after 2 days
0.5% after 5 days

0.5% is shown since 7 days, until now. 
So this sensor is currently useless. If there were the battery voltage exposed, it would be easier to analyze.

Before this TRVs, I had not-smart digital ones. And they simply shutting down without changing the valve position. 
This way it sometimes take one day or two to notice the empty batteries. But it prevents the transformation to a sauna within minutes. 😉

**Comment by magicbenny** [comment]
stange. i have 3 Spirits for about 4 weeks now. all
of them still show 100% battery state in Home Assistant. hmmmmm

**Comment by florianschroen** [comment]
Home Assistant shows only the transmitted battery states. But not the "not received states". 
Maybe you are affected by the bug where the TRV stops transmitting states after few hours / days. For me this seems to be 3 days, as you can see in the Screenshot. 

I am pushing the values from mqtt to volkszähler. There I can see the values. Some sensors are shown with dots; here mqtt_elapsed. This dots are an indicator for a new mqtt message from the TRV. Or zigbee2mqtt has restarted and reports the last known state to mqtt. 

![Screenshot_20210117_192730](https://user-images.githubusercontent.com/625094/104852255-27ef6200-58fa-11eb-9d8e-1b535abdcee2.jpg)

**Comment by docgalaxyblock** [comment]
I had last year a call with an technican from eurotronic because my thermostat got into error state when we were on holiday.
When I come back I had a bedroom themperature of near 30°C, opend the window the hole night and slept at on couch.
Also the part at the basement where the pipes are was a warmer then usual.

At the call the technican said for insurance reasons it opens the valve completely.
I understand this point and like to mention that a higher heating costs bill is better and cheaper as frozen pipes..

**Comment by Palcooky** [comment]
I have also experienced this. In a bedroom battery was dead valve was fully open. We had to remove the baby from the room. I have now replaced the bedroom with a normal thermostat due to the danger this could have had on the baby.

**Comment by drdownload** [comment]
Older TRV I had used to go to 25 % open if the battery was dead to prevent freezing. that would be good compromise
